map/mode.c

```diff
@@ -0,0 +1,81 @@
+#include "map/mode.h"
+#include "common/trinary/tryte_ascii.h"
+#include "mam/mam/mam_channel_t_set.h"
+
+retcode_t map_channel_create(mam_api_t *const api, tryte_t *const channel_id) {
+  if (mam_channel_t_set_size(api->channels) == 0) {
+    mam_api_channel_create(api, MSS_DEPTH, channel_id);
+  } else {
+    mam_channel_t *channel = &api->channels->value;
+    trits_to_trytes(trits_begin(mam_channel_id(channel)), channel_id, NUM_TRITS_ADDRESS);
+  }
+
+  return RC_OK;
+}
+
+retcode_t map_announce_channel(mam_api_t *const api, tryte_t const *const channel_id,
+                               bundle_transactions_t *const bundle, trit_t *const msg_id,
+                               tryte_t *const new_channel_id) {
+  retcode_t ret = RC_OK;
+
+  ERR_BIND_RETURN(mam_api_channel_create(api, MSS_DEPTH, new_channel_id), ret);
+
+  if ((ret = mam_api_bundle_announce_channel(api, channel_id, new_channel_id, NULL, NULL, bundle, msg_id)) != RC_OK) {
+    return ret;
+  }
```

Comment:
This if statement is redundant, `ret` will be returned in the next line.

Reply:
I wrap like this for later usage. It will have psk or ntru encryption in the future. But I guess it's no use here.